lib/doorkeeper/config.rb

```diff
@@ -321,6 +321,14 @@ def configure_secrets_for(type, using:, fallback:)
                grant_flows.exclude?("implicit")
            end)
 
+    # When using client credentials to authenticate at introspection endpoint,
+    # Allow introspection if the inspected token to be connected to the same client.
+    #
+    option :allow_token_introspection,
+           default: (lambda do |token, client|
+             token.application == client.application
```

Comment:
Actually, token can be issued without reference to application, so I think we need to add a guard clause like:

```ruby
return true unless token.application # Public token

token.application == client.application # Private token
```